sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -63,24 +68,56 @@ object CosmosAccountConfig {
       val url = new URL(accountEndpointUri)
       val separatorIndex = url.getHost.indexOf('.')
       if (separatorIndex > 0) {
-        url.getHost.substring(0, separatorIndex)
+          url.getHost.substring(0, separatorIndex)
       } else {
-        url.getHost()
+          url.getHost()
       }
     },
     helpMessage = "Cosmos DB Account Name")
 
+  private[spark] val ApplicationName = CosmosConfigEntry[String](key = "spark.cosmos.applicationName",
+    mandatory = false,
+    parseFromStringFunction = applicationName => applicationName,
+    helpMessage = "Application name")
+
+  private[spark] val UseGatewayMode = CosmosConfigEntry[Boolean](key = "spark.cosmos.useGatewayMode",
```

Comment:
CosmosConfigEntry has infra for supporting default value, could you please use that for passing the default.

default for UseGatewayMode should be false.

see sample please:
https://github.com/moderakh/azure-sdk-for-java/blob/eb13c04777dd8fb183f30e4b217e6c3bba573dcc/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala#L102

Reply:
Done